### PR TITLE
Parallel repeats and protection from infinite crash loop

### PR DESCRIPTION
**Pull request by RainNapper**
Cherry-picked changes from these two PRs.
https://github.com/linkedin/bluepill/pull/432
https://github.com/linkedin/bluepill/pull/431

### CODE REVIEW

File: bluepill/src/BPRunner.m [hunk 1]
```diff
@@ -213,24 +213,35 @@ - (int)runWithBPXCTestFiles:(NSArray<BPXCTestFile *> *)xcTestFiles {
...
-        numSims = bundles.count;
-    }
     if (self.config.cloneSimulator) {
         self.testHostForSimUDID = [bpSimulator createSimulatorAndInstallAppWithBundles:xcTestFiles];
         if ([self.testHostForSimUDID count] == 0) {
             return 1;
         }
     }
+
+    int repeatTestsCount = [self.config.repeatTestsCount intValue];
```
Comment (jerrymarino) [inline bluepill/src/BPRunner.m:223]: Won't this PR add repeat test count 2x?

Comment (RainNapper) [reply bluepill/src/BPRunner.m:223]: Oh, you're right. I think it was a bad merge. Updating.

**Comment by jerrymarino** [comment]
LGTM!

**Comment by jerrymarino** [comment]
@RainNapper is there any test coverage for this PR? I think it's going to end up running the same test over and over even if it didn't fail

**Pull request merged**